### PR TITLE
Sources

### LINKED ISSUE #176: `/search?layers` becomes Coarse
Search layers should be able offer mix & matched layers for different types of features (aliasing `admin0`, `admin1`, and `admin2` into `admin`, and aliasing `admin` into `coarse`).

New Aliases are created by the operator of the Pelias instance, not an API user nor an End User.

This is what lets us accomplish offering individual individual administrative layers as categories, while combining them to an `admin` option, as well as a `coarse` category replace /search/coarse.

### LINKED ISSUE #232: Layer options need to be updated
Currently if you type in an unexpected layer type you get this error message:

```
{
error: "invalid param 'layers': must be one or more of geoname,osmnode,osmway,admin0,admin1,admin2,neighborhood,locality,local_admin,osmaddress,openaddresses,poi,admin,address"
}
```

That's outdated. Layer options should be as follows:

* country
* region
* county
* neighbourhood
* locality
* localadmin
* address
* venue
* coarse (alias for all admin types combined)

### LINKED ISSUE #233: Unexpected `sources` type causes an exception
If you specify the wrong `sources` value, you'll get this:

```
{
error: "Cannot read property 'types' of undefined"
}
```

logs show:

```
2015-09-10_15:11:42.12987 2015-09-10T15:11:42.129Z - error: [middleware-500] Error: `TypeError: Cannot read property 'types' of undefined`. Stack trace: `TypeError: Cannot read property 'types' of undefined
2015-09-10_15:11:42.13000     at middleware (/srv/www/api/releases/20150909163117/middleware/_types.js:11:37)
2015-09-10_15:11:42.13006     at Layer.handle [as handle_request] (/srv/www/api/releases/20150909163117/node_modules/express/lib/router/layer.js:95:5)
2015-09-10_15:11:42.13010     at trim_prefix (/srv/www/api/releases/20150909163117/node_modules/express/lib/router/index.js:312:13)
2015-09-10_15:11:42.13014     at /srv/www/api/releases/20150909163117/node_modules/express/lib/router/index.js:280:7
2015-09-10_15:11:42.13017     at Function.process_params (/srv/www/api/releases/20150909163117/node_modules/express/lib/router/index.js:330:12)
2015-09-10_15:11:42.13020     at next (/srv/www/api/releases/20150909163117/node_modules/express/lib/router/index.js:271:10)
2015-09-10_15:11:42.13023     at /srv/www/api/releases/20150909163117/sanitiser/search.js:25:5
2015-09-10_15:11:42.13026     at sanitize (/srv/www/api/releases/20150909163117/sanitiser/_sanitize.js:8:14)
2015-09-10_15:11:42.13029     at sanitize (/srv/www/api/releases/20150909163117/sanitiser/search.js:12:36)
2015-09-10_15:11:42.13032     at module.exports.middleware (/srv/www/api/releases/20150909163117/sanitiser/search.js:19:3)`.
```

Add property error handling and an error message that lists possible values.

### LINKED ISSUE #237: Change `source` param to `sources`
For consistency.

### PR DESCRIPTION
**Pull request by dianashk**
* Renamed `source` to `sources`
* Refactored layers and sources into a common targets module

Fixes #237
Fixes #232
Fixes #176
Fixes #233

### CODE REVIEW

File: sanitiser/_targets.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+var _ = require('lodash');
+
+function setup(paramName, targetMap) {
```
Comment (dianashk) [inline sanitiser/_targets.js]: add comments here

File: sanitiser/_targets.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+function sanitize( paramName, targetMap, req ) {
+  var params = req.query || {};
+
+  req.clean = req.clean || {};
+  req.clean.types = req.clean.types || {};
+
+  // default case (no sources specified in GET params)
+  // there is a case where the property is present, but set to null or undefined
+  if (params.hasOwnProperty(paramName) === false || typeof params[paramName] === 'undefined') {
```
Comment (dianashk) [inline sanitiser/_targets.js]: check_types

File: sanitiser/_targets.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    return { error: false };
+  }
+
+  params[paramName] = params[paramName].trim();
```
Comment (dianashk) [inline sanitiser/_targets.js]: check for string type as well

File: sanitiser/_targets.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  if (invalidTargets.length > 0) {
+    return {
+      error: true,
+      message: invalidTargets[0] + ' is an invalid ' + paramName + ' parameter. Valid options: ' + Object.keys(targetMap).join(', ')
+    };
+  }
+
+  var cleanPropName = 'from_' + paramName;
```
Comment (dianashk) [inline sanitiser/_targets.js]: 'from_' can go away

**Comment by missinglink** [comment]
reviewed and [painfully] merged with master

**Pull request merged**